internal/services/springcloud/spring_cloud_elastic_application_performance_monitoring_resource.go

```diff
@@ -0,0 +1,310 @@
+)
+
+type SpringCloudElasticApplicationPerformanceMonitoringModel struct {
+	Name                 string   `tfschema:"name"`
+	SpringCloudServiceId string   `tfschema:"spring_cloud_service_id"`
+	GloballyEnabled      bool     `tfschema:"globally_enabled"`
+	ServiceName          string   `tfschema:"service_name"`
+	ApplicationPackages  []string `tfschema:"application_packages"`
+	ServerUrl            string   `tfschema:"server_url"`
+}
+
+type SpringCloudElasticApplicationPerformanceMonitoringResource struct{}
+
+var _ sdk.ResourceWithUpdate = SpringCloudElasticApplicationPerformanceMonitoringResource{}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) ResourceType() string {
+	return "azurerm_spring_cloud_elastic_application_performance_monitoring"
+}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) ModelObject() interface{} {
+	return &SpringCloudElasticApplicationPerformanceMonitoringModel{}
+}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return appplatform.ValidateApmID
+}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"spring_cloud_service_id": commonschema.ResourceIDReferenceRequiredForceNew(commonids.SpringCloudServiceId{}),
+
+		"application_packages": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+
+		"service_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"server_url": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"globally_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+	}
+}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (s SpringCloudElasticApplicationPerformanceMonitoringResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudElasticApplicationPerformanceMonitoringModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.AppPlatformClient
+			springId, err := commonids.ParseSpringCloudServiceID(model.SpringCloudServiceId)
+			if err != nil {
+				return fmt.Errorf("parsing spring service ID: %+v", err)
```

Comment:
The parsing functions already return sufficient information in the error messages, there is no need to wrap these. 
```suggestion
				return err
```